### PR TITLE
Backport: fix route name resolver cache with subresources

### PR DESCRIPTION
**Pull request by soyuka**
Prefering a cherry-pick of 3b53ff2e9041697b8a182beb21dd4ef4b29c3542
https://github.com/api-platform/core/pull/1770
https://github.com/api-platform/core/issues/1769

### CODE REVIEW

**Comment by dunglas** [comment]
For the record, we don't fix bugs anymore in 2.1. Only security fixes.

**Comment by soyuka** [comment]
> For the record, we don't fix bugs anymore in 2.1. Only security fixes.
 Delete branch

Yes I know that this is a one time exception because it's kinda blocking if you can't update :|.

File: src/Bridge/Symfony/Routing/CachedRouteNameResolver.php [hunk 1]
```diff
@@ -39,7 +39,8 @@ public function __construct(CacheItemPoolInterface $cacheItemPool, RouteNameReso
      */
     public function getRouteName(string $resourceClass, $operationType /**, array $context = []**/): string
     {
-        $cacheKey = self::CACHE_KEY_PREFIX.md5(serialize([$resourceClass, $operationType]));
+        $context = \func_num_args() > 2 ? func_get_arg(2) : [];
+        $cacheKey = self::CACHE_KEY_PREFIX.md5(serialize([$resourceClass, $operationType, $context['subresource_resources'] ?? null]));
```
Comment (meyerbaptiste) [inline src/Bridge/Symfony/Routing/CachedRouteNameResolver.php]: We have to add a test!

Comment (soyuka) [reply src/Bridge/Symfony/Routing/CachedRouteNameResolver.php]: RemindMe! 10 days (I wish there was such a bot here)

**Pull request merged**